File: 2145-grid-game/grid-game.java
class Solution {
    public long gridGame(int[][] grid) {
        int n = grid[0].length;
        List<Long> prefix_1 = new ArrayList<Long>();
        List<Long> prefix_2 = new ArrayList<Long>();

        for (int i: grid[0])
            prefix_1.add((long) i);

        for (int i: grid[1])
            prefix_2.add((long) i);

        for (int i = 1; i < n; i++){
            prefix_1.set(i, prefix_1.get(i) + prefix_1.get(i-1));
            prefix_2.set(i, prefix_2.get(i) + prefix_2.get(i-1));
        }

        long result = Long.MAX_VALUE;
        for (int i = 0; i < n; i++){
            long top = prefix_1.get(n - 1) - prefix_1.get(i);
            long bottom = i > 0 ? prefix_2.get(i - 1) : 0;
            long second = Math.max(top, bottom);
            result = Math.min(result, second);
        }
        return result;
    }
}

File: 2145-grid-game/grid-game.cpp
class Solution {
public:
    long long gridGame(vector<vector<int>>& grid) {
        int n = grid[0].size();
        vector<long long> prefix_1(grid[0].begin(), grid[0].end());
        vector<long long> prefix_2(grid[1].begin(), grid[1].end());

        for (int i = 1; i < n; i++){
            prefix_1[i] += prefix_1[i-1];
            prefix_2[i] += prefix_2[i-1];
        }

        long long result = LLONG_MAX;
        for (int i = 0; i < n; i++){
            long long top = prefix_1[n - 1] - prefix_1[i];
            long long bottom = i > 0 ? prefix_2[i - 1] : 0;
            long long second = max(top, bottom);
            result = min(result, second);
        }
        return result;
    }
};